cosmetics-web/app/models/notification.rb

```diff
@@ -315,6 +315,10 @@ def cloned?
     source_notification.present?
   end
 
+  def ingredients
+    components.map(&:ingredients).flatten
```

Comment:
To avoid N+1 queries for multicomponent notifications:
```suggestion
    components.includes(:ingredients).map(&:ingredients).flatten
```